src/Components/Shared/test/TestRenderer.cs

```diff
@@ -43,21 +40,8 @@ public new Task RenderRootComponentAsync(int componentId)
         public new Task RenderRootComponentAsync(int componentId, ParameterCollection parameters)
             => InvokeAsync(() => base.RenderRootComponentAsync(componentId, parameters));
 
-        public new void DispatchEvent(int componentId, int eventHandlerId, UIEventArgs args)
-        {
-            var t = Invoke(() => base.DispatchEvent(componentId, eventHandlerId, args));
-            // This should always be run synchronously
-            Assert.True(t.IsCompleted);
-            if (t.IsFaulted)
-            {
-                var exception = t.Exception.Flatten().InnerException;
-                while (exception is AggregateException e)
-                {
-                    exception = e.InnerException;
-                }
-                ExceptionDispatchInfo.Capture(exception).Throw();
-            }
-        }
+        public new Task DispatchEventAsync(int componentId, int eventHandlerId, UIEventArgs eventArgs)
+            => InvokeAsync(() => base.DispatchEventAsync(componentId, eventHandlerId, eventArgs));
```

Comment:
Is it necessary to pass the logic through `InvokeAsync` here? I'm concerned this might make the tests describe something different than how the real renderers would behave, since when dispatching events to the real renderers we don't force it through an async hop.

Reply:
Now that we require the use the sync context events have to be dispatched on the sync context. The renderer would already be running on the sync context when `DispatchEventAsync` is called.

So I guess I'm not sure what else this would do? 

Another *very intrusive* approach is a write a test framework that has a sync context, and then just ignore threading in your tests. We have done this a few places, but I don't really relish it here.  When you do this, a different set of test scenarios become hard.